library/src/main/java/com/orm/SugarRecord.java

```diff
@@ -85,6 +86,7 @@
             sqLiteDatabase.endTransaction();
             sqLiteDatabase.setLockingEnabled(true);
         }
+        return deletedRows;
```

Comment:
This return a non zero number if the transaction is aborted while it's occurring. However, since it's a transaction, it'll roll back all of the deletes, so you should be returning zero in that case.